### ISSUE TITLE
6700s

### ISSUE THREAD

**Issue by frabacc** [likelovewant/ollama-for-amd#15]
### What is the issue?

2024/09/11 01:57:28 routes.go:1125: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION:10.3.2 OLLAMA_DEBUG:false OLLAMA_FLASH_ATTENTION:false OLLAMA_GPU_OVERHEAD:0 OLLAMA_HOST:http://127.0.0.1:11434 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_LOAD_TIMEOUT:5m0s OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:C:\\Users\\Francesco\\.ollama\\models OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://*] OLLAMA_RUNNERS_DIR:C:\\Users\\Francesco\\AppData\\Local\\Programs\\Ollama\\lib\\ollama\\runners OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"
time=2024-09-11T01:57:28.185+02:00 level=INFO source=images.go:753 msg="total blobs: 5"
time=2024-09-11T01:57:28.191+02:00 level=INFO source=images.go:760 msg="total unused blobs removed: 0"
time=2024-09-11T01:57:28.193+02:00 level=INFO source=routes.go:1172 msg="Listening on 127.0.0.1:11434 (version 0.3.10-0-g486ae43)"
time=2024-09-11T01:57:28.195+02:00 level=INFO source=payload.go:44 msg="Dynamic LLM libraries [cpu cpu_avx cpu_avx2 rocm_v6.1]"
time=2024-09-11T01:57:28.196+02:00 level=INFO source=gpu.go:200 msg="looking for compatible GPUs"
time=2024-09-11T01:57:28.210+02:00 level=INFO source=gpu.go:347 msg="no compatible GPUs were discovered"
time=2024-09-11T01:57:28.210+02:00 level=INFO source=types.go:107 msg="inference compute" id=0 library=cpu variant=avx2 compute="" driver=0.0 name="" total="15.2 GiB" available="10.5 GiB"
[GIN] 2024/09/11 - 01:57:37 | 200 |       520.8µs |       127.0.0.1 | HEAD     "/"
[GIN] 2024/09/11 - 01:57:37 | 200 |     47.2589ms |       127.0.0.1 | POST     "/api/show"
time=2024-09-11T01:57:37.777+02:00 level=INFO source=server.go:101 msg="system memory" total="15.2 GiB" free="10.4 GiB" free_swap="11.0 GiB"
time=2024-09-11T01:57:37.778+02:00 level=INFO source=memory.go:326 msg="offload to cpu" layers.requested=-1 layers.model=33 layers.offload=0 layers.split="" memory.available="[10.4 GiB]" memory.gpu_overhead="0 B" memory.required.full="5.8 GiB" memory.required.partial="0 B" memory.required.kv="1.0 GiB" memory.required.allocations="[5.8 GiB]" memory.weights.total="4.7 GiB" memory.weights.repeating="4.3 GiB" memory.weights.nonrepeating="411.0 MiB" memory.graph.full="560.0 MiB" memory.graph.partial="677.5 MiB"
time=2024-09-11T01:57:37.783+02:00 level=INFO source=server.go:391 msg="starting llama server" cmd="C:\\Users\\Francesco\\AppData\\Local\\Programs\\Ollama\\lib\\ollama\\runners\\cpu_avx2\\ollama_llama_server.exe --model C:\\Users\\Francesco\\.ollama\\models\\blobs\\sha256-8eeb52dfb3bb9aefdf9d1ef24b3bdbcfbe82238798c4b918278320b6fcef18fe --ctx-size 8192 --batch-size 512 --embedding --log-disable --no-mmap --parallel 4 --port 49692"
time=2024-09-11T01:57:37.806+02:00 level=INFO source=sched.go:450 msg="loaded runners" count=1
time=2024-09-11T01:57:37.806+02:00 level=INFO source=server.go:590 msg="waiting for llama runner to start responding"
time=2024-09-11T01:57:37.807+02:00 level=INFO source=server.go:624 msg="waiting for server to become available" status="llm server error"
INFO [wmain] build info | build=3661 commit="8962422b" tid="7284" timestamp=1726012657
INFO [wmain] system info | n_threads=8 n_threads_batch=8 system_info="AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | SVE = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 0 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | " tid="7284" timestamp=1726012657 total_threads=16
INFO [wmain] HTTP server listening | hostname="127.0.0.1" n_threads_http="15" port="49692" tid="7284" timestamp=1726012657
llama_model_loader: loaded meta data with 29 key-value pairs and 292 tensors from C:\Users\Francesco\.ollama\models\blobs\sha256-8eeb52dfb3bb9aefdf9d1ef24b3bdbcfbe82238798c4b918278320b6fcef18fe (version GGUF V3 (latest))
llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.
llama_model_loader: - kv   0:                       general.architecture str              = llama
llama_model_loader: - kv   1:                               general.type str              = model
llama_model_loader: - kv   2:                               general.name str              = Meta Llama 3.1 8B Instruct
llama_model_loader: - kv   3:                           general.finetune str              = Instruct
llama_model_loader: - kv   4:                           general.basename str              = Meta-Llama-3.1
llama_model_loader: - kv   5:                         general.size_label str              = 8B
llama_model_loader: - kv   6:                            general.license str              = llama3.1
llama_model_loader: - kv   7:                               general.tags arr[str,6]       = ["facebook", "meta", "pytorch", "llam...
llama_model_loader: - kv   8:                          general.languages arr[str,8]       = ["en", "de", "fr", "it", "pt", "hi", ...
llama_model_loader: - kv   9:                          llama.block_count u32              = 32
llama_model_loader: - kv  10:                       llama.context_length u32              = 131072
llama_model_loader: - kv  11:                     llama.embedding_length u32              = 4096
llama_model_loader: - kv  12:                  llama.feed_forward_length u32              = 14336
llama_model_loader: - kv  13:                 llama.attention.head_count u32              = 32
llama_model_loader: - kv  14:              llama.attention.head_count_kv u32              = 8
llama_model_loader: - kv  15:                       llama.rope.freq_base f32              = 500000.000000
llama_model_loader: - kv  16:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010
llama_model_loader: - kv  17:                          general.file_type u32              = 2
llama_model_loader: - kv  18:                           llama.vocab_size u32              = 128256
llama_model_loader: - kv  19:                 llama.rope.dimension_count u32              = 128
llama_model_loader: - kv  20:                       tokenizer.ggml.model str              = gpt2
llama_model_loader: - kv  21:                         tokenizer.ggml.pre str              = llama-bpe
llama_model_loader: - kv  22:                      tokenizer.ggml.tokens arr[str,128256]  = ["!", "\"", "#", "$", "%", "&", "'", ...
llama_model_loader: - kv  23:                  tokenizer.ggml.token_type arr[i32,128256]  = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...
llama_model_loader: - kv  24:                      tokenizer.ggml.merges arr[str,280147]  = ["Ġ Ġ", "Ġ ĠĠĠ", "ĠĠ ĠĠ", "...
llama_model_loader: - kv  25:                tokenizer.ggml.bos_token_id u32              = 128000
llama_model_loader: - kv  26:                tokenizer.ggml.eos_token_id u32              = 128009
llama_model_loader: - kv  27:                    tokenizer.chat_template str              = {{- bos_token }}\n{%- if custom_tools ...
llama_model_loader: - kv  28:               general.quantization_version u32              = 2
llama_model_loader: - type  f32:   66 tensors
llama_model_loader: - type q4_0:  225 tensors
llama_model_loader: - type q6_K:    1 tensors
time=2024-09-11T01:57:38.061+02:00 level=INFO source=server.go:624 msg="waiting for server to become available" status="llm server loading model"
llm_load_vocab: special tokens cache size = 256
llm_load_vocab: token to piece cache size = 0.7999 MB
llm_load_print_meta: format           = GGUF V3 (latest)
llm_load_print_meta: arch             = llama
llm_load_print_meta: vocab type       = BPE
llm_load_print_meta: n_vocab          = 128256
llm_load_print_meta: n_merges         = 280147
llm_load_print_meta: vocab_only       = 0
llm_load_print_meta: n_ctx_train      = 131072
llm_load_print_meta: n_embd           = 4096
llm_load_print_meta: n_layer          = 32
llm_load_print_meta: n_head           = 32
llm_load_print_meta: n_head_kv        = 8
llm_load_print_meta: n_rot            = 128
llm_load_print_meta: n_swa            = 0
llm_load_print_meta: n_embd_head_k    = 128
llm_load_print_meta: n_embd_head_v    = 128
llm_load_print_meta: n_gqa            = 4
llm_load_print_meta: n_embd_k_gqa     = 1024
llm_load_print_meta: n_embd_v_gqa     = 1024
llm_load_print_meta: f_norm_eps       = 0.0e+00
llm_load_print_meta: f_norm_rms_eps   = 1.0e-05
llm_load_print_meta: f_clamp_kqv      = 0.0e+00
llm_load_print_meta: f_max_alibi_bias = 0.0e+00
llm_load_print_meta: f_logit_scale    = 0.0e+00
llm_load_print_meta: n_ff             = 14336
llm_load_print_meta: n_expert         = 0
llm_load_print_meta: n_expert_used    = 0
llm_load_print_meta: causal attn      = 1
llm_load_print_meta: pooling type     = 0
llm_load_print_meta: rope type        = 0
llm_load_print_meta: rope scaling     = linear
llm_load_print_meta: freq_base_train  = 500000.0
llm_load_print_meta: freq_scale_train = 1
llm_load_print_meta: n_ctx_orig_yarn  = 131072
llm_load_print_meta: rope_finetuned   = unknown
llm_load_print_meta: ssm_d_conv       = 0
llm_load_print_meta: ssm_d_inner      = 0
llm_load_print_meta: ssm_d_state      = 0
llm_load_print_meta: ssm_dt_rank      = 0
llm_load_print_meta: ssm_dt_b_c_rms   = 0
llm_load_print_meta: model type       = 8B
llm_load_print_meta: model ftype      = Q4_0
llm_load_print_meta: model params     = 8.03 B
llm_load_print_meta: model size       = 4.33 GiB (4.64 BPW) 
llm_load_print_meta: general.name     = Meta Llama 3.1 8B Instruct
llm_load_print_meta: BOS token        = 128000 '<|begin_of_text|>'
llm_load_print_meta: EOS token        = 128009 '<|eot_id|>'
llm_load_print_meta: LF token         = 128 'Ä'
llm_load_print_meta: EOT token        = 128009 '<|eot_id|>'
llm_load_print_meta: max token length = 256
llm_load_tensors: ggml ctx size =    0.14 MiB
llm_load_tensors:        CPU buffer size =  4437.81 MiB
llama_new_context_with_model: n_ctx      = 8192
llama_new_context_with_model: n_batch    = 512
llama_new_context_with_model: n_ubatch   = 512
llama_new_context_with_model: flash_attn = 0
llama_new_context_with_model: freq_base  = 500000.0
llama_new_context_with_model: freq_scale = 1
llama_kv_cache_init:        CPU KV buffer size =  1024.00 MiB
llama_new_context_with_model: KV self size  = 1024.00 MiB, K (f16):  512.00 MiB, V (f16):  512.00 MiB
llama_new_context_with_model:        CPU  output buffer size =     2.02 MiB
llama_new_context_with_model:        CPU compute buffer size =   560.01 MiB
llama_new_context_with_model: graph nodes  = 1030
llama_new_context_with_model: graph splits = 1
INFO [wmain] model loaded | tid="7284" timestamp=1726012661
time=2024-09-11T01:57:41.526+02:00 level=INFO source=server.go:629 msg="llama runner started in 3.72 seconds"

### OS

Windows

### GPU

AMD

### CPU

AMD

### Ollama version

0.3.10

**Comment by likelovewant** [comment]
Please follow the [wiki guide](https://github.com/likelovewant/ollama-for-amd/wiki#demo-release-version) , replace the rocmlibs for  gfx1032 as guided. @frabacc

**Comment by frabacc** [comment]
What I've done from your guide:

- Test Drive Ollama: If you want to test before building from source, download and install a demo release from ollama-for-amd/releases.
 - Replace Files: As instructed in the ROCm supprot, replace files in your Ollama program folder with the rocblas.dll and rocblas/library folder from the demo release that matches your GPU architecture.(eg. the file in （`C:\Users\usrname\AppData\Local\Programs\Ollama\lib\ollama,this report will not update regulary ,serve an example only .) Try pre-built ROCm libraries:

    Some pre-built rocblas libraries are available for specific GPUs at: ROCmLibs for HIP SDK 6.1.2
    Choose the appropriate library based on your GPU model (e.g., gfx902, gfx1032).

I've also downloaded the HIP SDK from AMD website and replaced the files as instructed, is it possible that i have to use gfx1031?

**Comment by likelovewant** [comment]
1.run ` hipinfo` in termnial, it will show your gpu info.whether `gfx1032 `or` gfx13x`. 6700s as the info  show [gfx1032](https://www.techpowerup.com/gpu-specs/radeon-rx-6700s.c3868).
2.make sure you had replace the rocblas.dll and update the latest drivers .
3. run the installer `OllamaSetup.exe` rather than the `7z file`. ( Had see someone misuse those file ,make it unrocoginzied or memory detection issue)

**Comment by likelovewant** [comment]
Closing due to inactivity

**Comment by frabacc** [comment]
Hi, sorry to bother you again, but with the newest update what I did was (from scratch)
- I installed the OllamaSetup.exe file, version 0.4.1, from your repository.
- Replaced the files in the Ollama directory, rocblas.dll and the rocblas/library folder
- Installed the HIP SDK 5.7.1 and replaced the rocblas/library folder and the rocblas.dll
- Add `HSA_OVERRIDE_GFX_VERSION=10.3.0` as a system enviroment variable 

This is the Ollama log
```
2024/11/13` 00:25:32 routes.go:1189: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION:10.3.0 HTTPS_PROXY: HTTP_PROXY: NO_PROXY: OLLAMA_DEBUG:false OLLAMA_FLASH_ATTENTION:false OLLAMA_GPU_OVERHEAD:0 OLLAMA_HOST:http://127.0.0.1:11434 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_LOAD_TIMEOUT:5m0s OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:C:\\Users\\user\\.ollama\\models OLLAMA_MULTIUSER_CACHE:false OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://* vscode-webview://*] OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"
time=2024-11-13T00:25:32.295+01:00 level=INFO source=images.go:755 msg="total blobs: 0"
time=2024-11-13T00:25:32.295+01:00 level=INFO source=images.go:762 msg="total unused blobs removed: 0"
time=2024-11-13T00:25:32.295+01:00 level=INFO source=routes.go:1240 msg="Listening on 127.0.0.1:11434 (version 0.4.1-1-g14a68a0)"
time=2024-11-13T00:25:32.296+01:00 level=INFO source=common.go:49 msg="Dynamic LLM libraries" runners="[cpu cpu_avx cpu_avx2 rocm]"
time=2024-11-13T00:25:32.296+01:00 level=INFO source=gpu.go:221 msg="looking for compatible GPUs"
time=2024-11-13T00:25:32.296+01:00 level=INFO source=gpu_windows.go:167 msg=packages count=1
time=2024-11-13T00:25:32.296+01:00 level=INFO source=gpu_windows.go:214 msg="" package=0 cores=8 efficiency=0 threads=16
time=2024-11-13T00:25:32.304+01:00 level=INFO source=gpu.go:386 msg="no compatible GPUs were discovered"
time=2024-11-13T00:25:32.304+01:00 level=INFO source=types.go:123 msg="inference compute" id=0 library=cpu variant=avx2 compute="" driver=0.0 name="" total="15.2 GiB" available="10.5 GiB"
[GIN] 2024/11/13 - 00:29:43 | 200 |      1.0608ms |       127.0.0.1 | HEAD     "/"
```
